### PR TITLE
By branch

### PR DESCRIPTION
**Pull request by AntonPanchishko**

### CODE REVIEW

File: src/main/java/core/basesyntax/controler/ConsoleHandler.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+public class ConsoleHandler {
+    BetDao betDao = new BetDaoImpl();
+    UserDao userDao = new UserDaoImpl();
+
+    public void betHandler() {
+        Scanner scanner = new Scanner(System.in);
+
+        while (true) {
+            String command = scanner.nextLine();
+            if (command.equalsIgnoreCase("q")) {
```
Comment (vicire) [inline src/main/java/core/basesyntax/controler/ConsoleHandler.java]: make "q" constant

File: src/main/java/core/basesyntax/controler/ConsoleHandler.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+                return;
+            }
+            Bet bet = null;
+            try {
+                String[] split = command.split(" ");
```
Comment (vicire) [inline src/main/java/core/basesyntax/controler/ConsoleHandler.java]: Same here

File: src/main/java/core/basesyntax/controler/ConsoleHandler.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+                int value = Integer.parseInt(split[0]);
```
Comment (vicire) [inline src/main/java/core/basesyntax/controler/ConsoleHandler.java]: here

File: src/main/java/core/basesyntax/controler/ConsoleHandler.java [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+                double risk = Double.parseDouble(split[1]);
```
Comment (vicire) [inline src/main/java/core/basesyntax/controler/ConsoleHandler.java]: and here

File: src/main/java/core/basesyntax/controler/ConsoleHandler.java [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+                bet = new Bet(value, risk);
+            } catch (NumberFormatException e) {
```
Comment (vicire) [inline src/main/java/core/basesyntax/controler/ConsoleHandler.java]: what happened if user enter 1 String parameter in command line?

File: src/main/java/core/basesyntax/controler/ConsoleHandler.java [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+                System.out.println("Будь-ласка введіть коректні дані");
```
Comment (vicire) [inline src/main/java/core/basesyntax/controler/ConsoleHandler.java]: Use English

File: src/main/java/core/basesyntax/controler/ConsoleHandler.java [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+            betDao.addBet(bet);
+            System.out.println(bet);
+        }
+    }
+
+    public void userHandler() {
+        Scanner scanner = new Scanner(System.in);
+        while (true) {
+            String command = scanner.nextLine();
+            if (command.equalsIgnoreCase("q")) {
```
Comment (vicire) [inline src/main/java/core/basesyntax/controler/ConsoleHandler.java]: Same remarks as in betHandler

File: src/main/java/core/basesyntax/controler/ConsoleHandler.java [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+import core.basesyntax.dao.BetDao;
+import core.basesyntax.dao.BetDaoImpl;
+import core.basesyntax.dao.UserDao;
+import core.basesyntax.dao.UserDaoImpl;
+import core.basesyntax.model.Bet;
+import core.basesyntax.model.User;
+import java.util.Scanner;
+
+public class ConsoleHandler {
+    BetDao betDao = new BetDaoImpl();
```
Comment (vicire) [inline src/main/java/core/basesyntax/controler/ConsoleHandler.java]: add access modifiers

**Review by vicire — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/controler/ConsoleHandler.java [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+            try {
+                String[] split = command.split(DELIMITER);
+                String name = split[NAME_INDEX];
+                String lastname = split[LASTNAME_INDEX];
+                if (name.equals("")|| lastname.equals("")) {
+                    System.out.println("Please enter correct name and lastname");
+                    continue;
+                }
+                user = new User(name, lastname);
+            } catch (NumberFormatException e) {
```
Comment (vicire) [inline src/main/java/core/basesyntax/controler/ConsoleHandler.java]: user can enter 1 String here too))

**Review by vicire — APPROVED** [review]

**Review by dmytro-hryhoruk — APPROVED** [review]

File: src/main/java/core/basesyntax/lib/Injector.java [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+                        || !(Factory.getUserDao().getClass()
+                        .isAnnotationPresent(Dao.class)));
+                if (predicate.test(field)) {
+                    throw new RuntimeException("No such class annotation");
+                }
+                if (field.getType() == UserDao.class) {
+                    field.set(instance, Factory.getUserDao());
+                } else if (field.getType() == BetDao.class) {
+                    field.set(instance, Factory.getBetDao());
+                }
```
Comment (Max-Pochepets) [inline src/main/java/core/basesyntax/lib/Injector.java]: I think u misunderstood the task.
1. Find the fields of the class, which being injected, for `@Inject`
2. Check the `Type` of this fields for `@Dao`

File: src/main/java/core/basesyntax/lib/Injector.java [hunk 11]
```diff
@@ -0,0 +1,42 @@
...
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+import java.util.function.Predicate;
+
+public class Injector implements Inject {
+    @Override
+    public Class<? extends Annotation> annotationType() {
+        return null;
+    }
```
Comment (Max-Pochepets) [inline src/main/java/core/basesyntax/lib/Injector.java]: whats this and were is it used?

File: src/main/java/core/basesyntax/lib/Dao.java [hunk 12]
```diff
@@ -0,0 +1,4 @@
+package core.basesyntax.lib;
+
+public @interface Dao {
+}
```
Comment (Max-Pochepets) [inline src/main/java/core/basesyntax/lib/Dao.java]: does this work?

File: src/main/java/core/basesyntax/Main.java [hunk 13]
```diff
@@ -1,11 +1,19 @@
...
-        ConsoleHandler consoleHandler = new ConsoleHandler();
+    public static void main(String[] args) throws InvocationTargetException,
+            NoSuchMethodException, InstantiationException, IllegalAccessException {
+        ConsoleHandler consoleHandler = (ConsoleHandler) Injector
+                .getInstance(ConsoleHandler.class);
         System.out.println("Enter value and risk");
-        consoleHandler.userHandler();
+        consoleHandler.betHandler();
+        BetDao betDao = Factory.getBetDao();
+        System.out.println("All bets: " + betDao.getAllBets());
```
Comment (Max-Pochepets) [inline src/main/java/core/basesyntax/Main.java]: if u have both the two storages for Bet and User respectively, as well as two Dao implementations, you should use both and then output the result

File: src/main/java/core/basesyntax/lib/Injector.java [hunk 14]
```diff
@@ -0,0 +1,42 @@
...
+        Field[] declaredFields = clazz.getDeclaredFields();
+        for (Field field: declaredFields) {
+            if (field.getAnnotation(Inject.class) != null) {
+                field.setAccessible(true);
+                Predicate predicate = o -> !(Factory.getBetDao().getClass()
+                        .isAnnotationPresent(Dao.class)
+                        || !(Factory.getUserDao().getClass()
+                        .isAnnotationPresent(Dao.class)));
+                if (predicate.test(field)) {
+                    throw new RuntimeException("No such class annotation");
```
Comment (Max-Pochepets) [inline src/main/java/core/basesyntax/lib/Injector.java]: we should throw custom exception

**Review by Max-Pochepets — CHANGES_REQUESTED** [review]
You should have created a new branch, not pushing to the same one

**Review by Max-Pochepets — APPROVED** [review]

**Comment by RomanDubovskyi** [comment]
PR should not containt files from previous hw, fix that before review

**Pull request merged**